lib/beaker/shared/subcommands_util.rb

```diff
@@ -30,7 +41,54 @@ def reset_argv(args)
       # @param [Array<String>] args the arguments determined by a specific subcommand
       def execute_beaker(*args)
         reset_argv(args)
-        Beaker::CLI.new.execute! 
+        Beaker::CLI.new.execute!
+      end
+
+      # Check for the presence of a Rakefile containing the require of the
+      # quick start tasks
+      def require_tasks()
+        unless File.file?(RAKE_FILE)
+          File.open RAKE_FILE, "w"
+        end
+        unless File.readlines(RAKE_FILE).grep(/#{BEAKER_REQUIRE}/).any?
+          File.open(RAKE_FILE,"a+") { |f| f.puts(BEAKER_REQUIRE) }
+        end
+      end
+
+      # Clear ARGV and execute a Rake task
+      # @param [String] task the rake task to execute
+      def execute_rake_task(task)
+        ARGV.clear
+        rake = Rake.application
+        rake.init()
+        rake.load_rakefile()
+        rake.invoke_task(task)
+      end
+
+      def init_vagrant()
+        execute_rake_task("beaker_quickstart:gen_hosts[vagrant]")
+      end
+
+      def init_vmpooler()
+        execute_rake_task("beaker_quickstart:gen_hosts[vmpooler]")
+      end
+
+      # Print a message to the console and exit with 0
+      # @param [String] msg the message to print
+      def exit_with(msg)
+        puts msg
+        exit(0)
+      end
+
+      # Verify that either the vagrant or vmpooler hypervisor has been specified
+      # @param [Array<Object>] options the options we can to query
+      def verify_init_args(options)
+        if options[:vmpooler] && options[:vagrant]
+          exit_with("You must specify only one hypervior, --vmpooler or --vagrant")
+        end
+        unless options[:vmpooler] || options[:vagrant]
+          exit_with("You must specify a hypervisor with `beaker init --vagrant` or `beaker init --vmpooler`")
+        end
```

Comment:
I don't like this method, but I need a way to assert that the user hasn't called init for both vmpooler and vagrant, but that they have specified at least one hypervisor

It's not possible to do this with Thor, as the traditional route would be to mark the flags as required, and we can't mark them both as required, so I'm having to perform a manual check here